From a0d3df3201b1af26868f03ca1e786bcf4cb8ab22 Mon Sep 17 00:00:00 2001
From: Tobias Kahan <tobyk100@uw.edu>
Date: Thu, 1 Nov 2012 16:08:41 -0700
Subject: [PATCH 11/17] debugging connectNeighbors

---
 connectNeighbors.py | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/connectNeighbors.py b/connectNeighbors.py
index a50dbd5..90612d6 100755
--- a/connectNeighbors.py
+++ b/connectNeighbors.py
@@ -6,12 +6,8 @@ SEPARATOR = '\t'  # The character separating nodes in a line of input
 def main(source):
   edges = read_graph(source)
   graphs = parse_graph(edges)
-  print(graphs[0])
-  print(graphs[1])
   remove_duplicates(graphs)
-  print(graphs[0])
-  print(graphs[1])
-  print_results(graphs)
+  print_results(graphs[0])
 
 def read_graph(source): 
   # Iteratre through each line of stdin, stripping trailing whitespace and
@@ -53,7 +49,10 @@ def check_remove_node(ingraph, outgraph, node):
           ingraph.pop(inNode)
           outgraph.pop(outNode)
 
-def print_results(graphs):
+def print_results(graph):
+  for k, v in graph.items():
+    for node in v:
+      print k + SEPARATOR + node
 
 
 if __name__ == "__main__":
-- 
1.7.11.7

